ZhitnikovTarasHomework3

Toptimum opened the pull request:
Решил все задачи.

Toptimum committed:
Решил все задачи.

DmitryChitalov commented on Lesson_3/1.py:
```diff
@@ -1,2 +1,15 @@
+        if number_series[j] % multiple_numbers[i] == 0:
+            sum += 1
+            list_multiple.append(number_series[j])
+    print(f"Числу {multiple_numbers[i]} кратно {sum} чисел: {list_multiple}.")
```
отлично, очень подробное решения

DmitryChitalov commented on Lesson_3/2.py:
```diff
@@ -4,4 +4,13 @@
+    if int(number_series[i]) % 2 == 0:
+        indexes.append(i + 1)
+
+print(f"В списке чисел {number_series} четные элементы находятся на следующих позициях: {indexes}.")
```
тут надо перепроверить
Введите список чисел через пробел( ): 4 4 5 6
В списке чисел ['4', '4', '5', '6'] четные элементы находятся на следующих позициях: [1, 2, 4].

DmitryChitalov commented on Lesson_3/3.py:
```diff
@@ -1,2 +1,28 @@
+# теперь можем поменять элементы местами
+number_series[min_number_index] = max_number
+number_series[max_number_index] = min_number
+print(f"Список с поменянными минимальным и максимальным элементами: {number_series}.")
```
отрабатывает верно

DmitryChitalov commented on Lesson_3/4.py:
```diff
@@ -1 +1,20 @@
+    if quantity > max_quantity:  # запоминаем самое популярное число
+        max_quantity = quantity
+        number = number_series[i]
+print(f"В списке чаще всего встречается число '{number}' - оно повторяется {max_quantity} раз(а).")
```
все верно

DmitryChitalov commented on Lesson_3/5.py:
```diff
@@ -1,2 +1,16 @@
+    if number_series[i] < min_number:  # находим минимальное значение и запоминаем его
+        min_number = number_series[i]
+        min_number_index = i
+print(f"В списке минимальное число '{min_number}' находится по индексу {min_number_index}.")
```
мы решили, что макс. отриц это ближе к 0

DmitryChitalov commented on Lesson_3/6.py:
```diff
@@ -1,5 +1,43 @@
+        sum_numbers += range_min_max[i]
+    print(f"Сумма элементов = {sum_numbers}.")
+else:
+    print(f"К сожалению, в списке между минимальным и максимальным числами нет элементов.")
```
все верно

DmitryChitalov commented on Lesson_3/7.py:
```diff
@@ -3,3 +3,19 @@
+    if number_series[i] <= min_number1:
+        min_number2 = min_number1
+        min_number1 = number_series[i]
+print(f"В списке минимальные числа '{min_number1}' и '{min_number2}'.")
```
надо доработать
Список случайно сгенерированных чисел: [9, 9, 10, 1, 10, 5, 11, 15, 2, 5, 4, 13, 13, 11, 7]
В списке минимальные числа '1' и '9'.

DmitryChitalov commented on Lesson_3/8.py:
```diff
@@ -4,3 +4,26 @@
+        print(array1[i][j], end=' ')
+        sum += array1[i][j]
+    print(f"= {sum}", end=' ')
+    print()
```
здесь бы хорошо не по одном числу запрашивать
а хотя бы строкой
иначе тестить очень тяжело

DmitryChitalov commented on Lesson_3/9.py:
```diff
@@ -1 +1,21 @@
+        numbers_column.append(number_array[i][j])  # собираем элементы столбцов в список
+    numbers_column.sort()  # сортируем элементы столбца и тогда на втором месте будет второе минимальное число
+    print(f"В {j + 1} столбце максимальный элемент среди минимальных является число '{numbers_column[1]}'")
+    numbers_column = []
```
все хорошо

Toptimum committed:
Решил все задачи.

Toptimum committed:
Решил все задачи.
В файле 2.py ручное сложение двух чисел, без реализации умножения.
В файле 2_easy.py простое решение задачи (и сложение и умножение).
На протяжении решения этих задач (2 дня), изучал Collections и пытался их применить. К сожалению, я не нашел способ их применить. Для меня это не очевидно.
